### ISSUE TITLE
How to skip parameter scope in oauth2 Authorization code flow

### ISSUE THREAD

**Issue by dinurp** [AnWeber/httpyac#560]
Is here a way to skip the parameter `scope` when requesting authorization code in the flow `code` for oauth2? Currently, if scope is not specified, `openid` is set as the scope. I would like to request all the scopes the user is authorized for. This can be requested by not specifying scope when the authorization code is requested. Currently I am unable to get the request to skip sending scope. The behavior I am looking for is same as that is currently in place for `client_credentials` flow. With this flow no default scope is requested when none is specified.  

I see in [plugins/oauth2/flow/authorizationCodeFlow.ts#L38](https://github.com/AnWeber/httpyac/blob/main/src/plugins/oauth2/flow/authorizationCodeFlow.ts#L38)
```
          }${utils.toQueryParams({
            client_id: config.clientId,
            scope: config.scope || 'openid',    <-- ****here****
            response_type: 'code',
            state,
            audience: config.audience,
            resource: config.resource,
            redirect_uri: config.redirectUri.toString(),
            ...(code_verifier
              ? { code_challenge: this.createSha256(code_verifier), code_challenge_method: 'S256' }
              : {}),
          })}`;
```
and in [ toQueryParams ](https://github.com/AnWeber/httpyac/blob/main/src/utils/requestUtils.ts#L161)
```
export function toQueryParams(params: Record<string, undefined | string | number | boolean>): string {
  return Object.entries(params)
    .filter(([, value]) => !!value)      <-- ****here****
    .map(([key, value]) => `${key}=${encodeURIComponent(value || '')}`)
    .join('&');
}
```
I could not see what I could set the value of `scope` as so that I could skip defaulting to `openid` but get filtered out in `toQueryParams`.

**Comment by AnWeber** [comment]
The idea is that I make a distinction between undefined and null/string empty. That is, if it is undefined, the logic takes effect as before. However, if the value is `""`, no explicit value is set. This makes it possible to override the value explicitly. for example using .env

```
oauth2_tokenEndpoint=http://localhost:8080/realms/demo/protocol/openid-connect/token
oauth2_scope=
```

**Comment by AnWeber** [comment]
will be released with next version

**Comment by dinurp** [comment]
Thanks @AnWeber. This fix in 6.8.0 provides the solution.
